naïve_bayes/sentiment_analysis/model.py

```diff
@@ -0,0 +1,159 @@
+        self._save(word_freq, class_freq)
+
+    def _save(self, word_freq: Dict[str, Dict[int, int]], class_freq: Dict[int, int]) -> None:
+        """
+        Save parameters to JSON file.
+
+        :param word_freq: Dict of words frequency per class
+        :param class_freq: Dict of quantity of tweets per class
+        :return:
+        """
+        general_dict = {'word_freq': word_freq, 'class_freq': class_freq}
+
+        try:
+            with open(self.file_path, 'w') as file:
+                file.write(json.dumps(general_dict))
+        except FileNotFoundError:
+            logging.error('File model')
+
+    def _import_params(self) -> Tuple[Dict[str, Dict[str, int]], Dict[str, int]]:
+        """
+        Import parameters from JSON file.
+
+        :param file_path: Path to file with parameters
+        :return: Tuple with 2 dicts:
+                - word_freq (words frequency per class)
+                - class_freq (quantity of tweets per class)
+        """
+        try:
+            trained_model = open(self.file_path)
+        except FileNotFoundError:
+            logging.error('File with model parameters has not been found, please train model!')
+            sys.exit(1)
+        except OSError as error:
+            logging.error(f'OS Error: {error}')
+            sys.exit(1)
+
+        parameters = json.load(trained_model)
+
+        try:
+            return parameters['word_freq'], parameters['class_freq']
+        except KeyError:
+            logging.error('Incorrect content in parameters file')
+            sys.exit()
+
+    def predict(self, tweets: pd.Series) -> pd.Series:
+        """
+        Predict is tweet positive/negative.
+
+        :param tweets: pd.Series of tweets with text
+        :param file_path: Path to parameters file
+        :return: pd.Series of predictions
+        """
+        predictions = []
+
+        word_freq, class_freq = self._import_params()
+
+        for text in tweets:
+            prob_dict = {0: 1.0, 1: 1.0}
+
+            for word in text:
+                if word in word_freq.keys():
+                    for i in range(2):
+                        prob_dict[i] *= word_freq[word][str(i)] / class_freq[str(i)]
+
+            for i in range(2):
+                prob_dict[i] *= class_freq[str(i)]
+
+            predictions.append(max(prob_dict, key=prob_dict.get))
+
+        return pd.Series(predictions, index=tweets.index)
+
+    def evaluate(self) -> float:
+        """Evaluate model accuracy."""
+        predictions = self.predict(self.test_data['text'])
+        miss_count = 0
+        correct_count = 0
+
+        for predict, sentiment in zip(predictions, self.test_data['sentiment']):
+            if predict != sentiment:
+                miss_count += 1
+            else:
+                correct_count += 1
+
+        accuracy = correct_count * 100 / len(self.test_data)
+
+        logging.info(f'Model accuracy is: {accuracy}%')
+        return accuracy
+
+
+def main() -> None:
```

Comment:
Maybe create a main.py file with methods for using the model that would be easy to build into the website

Reply:
For training a model, user will just run a model.py file, for making a prediction - import class. 
I don't see necessity to create a main.py file.